resources/aws/keypair.go

```diff
@@ -0,0 +1,79 @@
+package aws
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	microerror "github.com/giantswarm/microkit/error"
+
+	awsclient "github.com/giantswarm/aws-operator/client/aws"
+)
+
+type KeyPairProvider interface {
+	pubKeyContent() ([]byte, error)
+}
+
+type FSKeyPairProvider struct {
+	pubKeyFile string
+}
+
+func NewFSKeyPairProvider(pubKeyFile string) *FSKeyPairProvider {
+	return &FSKeyPairProvider{
+		pubKeyFile: pubKeyFile,
+	}
+}
+
+func (f *FSKeyPairProvider) pubKeyContent() ([]byte, error) {
+	return ioutil.ReadFile(f.pubKeyFile)
```

Comment:
Sure, but if your concern is to not rely unit tests on FS, then this Provider interface solves the problem.